data_model/primitives/src/lib.rs

```diff
@@ -15,3 +15,91 @@ extern crate alloc;
 pub mod atomic;
 pub mod fixed;
 pub mod small;
+
+use fixed::prelude::*;
+
+/// Trait for values that could be converted to `f64` and observed by prometheus.
+pub trait FloatMetricFrom {
+    /// Convert `value` to `f64`.
+    fn from(value: Self) -> f64;
+}
+
+impl FloatMetricFrom for u128 {
+    #[inline]
+    #[allow(clippy::cast_precision_loss)]
+    fn from(value: Self) -> f64 {
+        value as f64
+    }
+}
+
+impl FloatMetricFrom for u32 {
+    #[inline]
+    fn from(value: Self) -> f64 {
+        value.into()
+    }
+}
+
+impl FloatMetricFrom for Fixed {
+    #[inline]
+    fn from(value: Self) -> f64 {
+        value.into()
+    }
+}
+
+/// Trait for `checked_add` operation on iroha's primitive types.
+pub trait CheckedAdd {
+    /// Checked addition. Computes `self + rhs`, returning None if overflow occurred.
+    fn checked_add(self, rhs: Self) -> Option<Self>
+    where
+        Self: Sized;
+}
+
+impl CheckedAdd for u32 {
+    #[inline]
+    fn checked_add(self, rhs: Self) -> Option<Self> {
+        self.checked_add(rhs)
+    }
+}
+
+impl CheckedAdd for u128 {
+    #[inline]
+    fn checked_add(self, rhs: Self) -> Option<Self> {
+        self.checked_add(rhs)
+    }
+}
+
+impl CheckedAdd for Fixed {
+    #[inline]
+    fn checked_add(self, rhs: Self) -> Option<Self> {
+        self.checked_add(rhs).ok()
```

Comment:
Looks like it will be good to unify `Fixed::checked_add()` with types from `std`. Although it will stil require to use this trait...

Reply:
Yes, we still need this trait, and `Fixed::checked_add` can fail for several reasons, so if we re-implement `checked_add` we will lose the source of the error.